internal/modgen/modgen.go

```diff
@@ -0,0 +1,420 @@
+package modgen
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"math"
+	"math/rand"
+	"strconv"
+
+	"github.com/tetratelabs/wazero/internal/leb128"
+	"github.com/tetratelabs/wazero/internal/wasm"
+)
+
+// Gen generates a pseudo random compilable module based on `seed`. The size of
+// the generated module corresponds to the size of the seed. For example,
+// the more larger len(seed) is, the more functions the module *likely* has.
+//
+// TODO: make it possible to explicitly control the size of each section.
+// TODO: have an option to have whether or not the result has imports.
+//
+// Note: "pseudo" here means the determinism of the generated results,
+// e.g. giving same seed returns exactly the same module for
+// the same code base in Gen.
+//
+// Note: this is only used for testing wazero runtime.
+func Gen(seed []byte) *wasm.Module {
```

Comment:
gen should take a features thing I think, so that it can control which features it will generate.